File: 125-valid-palindrome/valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        //two pointers
        int left = 0; //start
        int right = s.length() -1; //end

        while(left < right)
        {
             //skip non alphanumeric character
             while(left < right && !Character.isLetterOrDigit(s.charAt(left))){
                left++;
             }
             while(left < right && !Character.isLetterOrDigit(s.charAt(right))){
                right--;
             }
             //move pointers on match else return false
             if(Character.toLowerCase(s.charAt(left)) != Character.toLowerCase(s.charAt(right))){
                return false;
             }
             left++;
             right--;
        }
        return true;
    }
}

File: 125-valid-palindrome/valid-palindrome.py
class Solution(object):
    def isPalindrome(self, s):
        left = 0
        right = len(s) - 1
        while left < right :
            # Skip non-alphanumeric characters from the left
            while left < right and not s[left].isalnum() :
                left += 1
            
            # Skip non-alphanumeric characters from the right
            while left < right and not s[right].isalnum() :
                right -= 1

            # compare and move forward
            if s[left].lower() != s[right].lower() :
                return False
            left += 1
            right -= 1
        return True
        